static/app/components/events/interfaces/frame/stacktraceLink.tsx

```diff
@@ -246,69 +221,27 @@ class StacktraceLink extends AsyncComponent<Props, State> {
     );
   }
 
-  renderNoIntegrations() {
-    const {organization} = this.props;
-    return (
-      <CodeMappingButtonContainer columnQuantity={2}>
-        <StyledLink to={`/settings/${organization.slug}/integrations/`}>
-          <StyledIconWrapper>{getIntegrationIcon('github', 'sm')}</StyledIconWrapper>
-          {t(
-            'Add a GitHub, Bitbucket, or similar integration to make sh*t easier for your team'
-          )}
-        </StyledLink>
-        <CloseButton type="button" priority="link" onClick={this.dismissPrompt}>
-          <IconClose size="xs" aria-label={t('Close')} />
-        </CloseButton>
-      </CodeMappingButtonContainer>
-    );
-  }
-
-  renderLink() {
-    const {config, sourceUrl} = this.state.match;
-    const url = `${sourceUrl}#L${this.props.frame.lineNo}`;
-    return (
-      <CodeMappingButtonContainer columnQuantity={2}>
-        <OpenInLink onClick={this.onOpenLink} href={url} openInNewTab>
-          <StyledIconWrapper>
-            {getIntegrationIcon(config!.provider.key, 'sm')}
-          </StyledIconWrapper>
-          {t('Open this line in %s', config!.provider.name)}
-        </OpenInLink>
-      </CodeMappingButtonContainer>
-    );
+  // No integrations, but prompt is dismissed or hidden
+  if (hideErrors || promptDismissed) {
+    return null;
   }
 
-  renderBody() {
-    const {config, error, sourceUrl, integrations} = this.state.match || {};
-    const {isDismissed, promptLoaded} = this.state;
-    const {event, line} = this.props;
-
-    // Success state
-    if (config && sourceUrl) {
-      return this.renderLink();
-    }
-
-    // Hide stacktrace link errors if the stacktrace might be minified javascript
-    // Check if the line starts and ends with {snip}
-    const hideErrors = event.platform === 'javascript' && /(\{snip\}).*\1/.test(line);
-
-    // Code mapping does not match
-    // Has integration but no code mappings
-    if (!hideErrors && (error || integrations.length > 0)) {
-      return this.renderNoMatch();
-    }
-
-    if (hideErrors || !promptLoaded || (promptLoaded && isDismissed)) {
-      return null;
-    }
-
-    return this.renderNoIntegrations();
-  }
+  // No integrations
+  return (
```

Comment:
The logic might be simpler if this were extracted to its own component and we put the `usePromptsCheck` in there, since it only ever matters if we get to this point, right? If you're worried about the request time you could put an unused `usePromptsCheck()` at the top of this component